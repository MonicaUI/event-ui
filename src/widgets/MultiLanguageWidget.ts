import { connect } from 'react-redux';
import MultiLanguageWidget from 'event-widgets/lib/MultiLanguage/MultiLanguageWidget';
import { loadLanguage } from '../redux/multiLanguage/actions';

/**
 * Data wrapper for the multi language widget
 */
export default connect(
  (state: $TSFixMe) => {
    return {
      locales: state.event.eventLocalesSetup.eventLocales,
      hideWhenOneLanguage: true
    };
  },
  { loadLanguage }
)(MultiLanguageWidget);
