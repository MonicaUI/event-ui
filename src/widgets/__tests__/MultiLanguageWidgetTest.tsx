import React from 'react';
import MultiLanguageWidget from '../MultiLanguageWidget';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

function getState() {
  return {
    locales: [],
    currentLocale: 'en-US',
    event: {
      eventLocalesSetup: {
        eventLocales: [
          {
            localeId: 1033,
            languageName: 'English',
            cultureCode: 'en-US'
          }
        ]
      }
    }
  };
}

const middlewares = [thunk];

const defaultProps = {
  translate: c => c,
  style: {},
  classes: {},
  multiLanguageLocale: '',
  loadLanguage: c => c,
  metaData: {}
};

describe('MultiLanguageWidget', () => {
  test('should match snapshot', () => {
    const widget = shallow(
      // @ts-expect-error ts-migrate(2322) FIXME: Type '{ store: any; translate: (c: any) => any; st... Remove this comment to see the full error message
      <MultiLanguageWidget {...defaultProps} store={configureMockStore(middlewares)(getState())} />
    );
    expect(widget.props()).toMatchSnapshot();
  });
});
