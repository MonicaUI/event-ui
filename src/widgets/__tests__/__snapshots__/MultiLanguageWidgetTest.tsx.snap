// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiLanguageWidget should match snapshot 1`] = `
Object {
  "children": <Memo(Connect(MultiLanguageWidget))
    classes={Object {}}
    hideWhenOneLanguage={true}
    loadLanguage={[Function]}
    locales={
      Array [
        Object {
          "cultureCode": "en-US",
          "languageName": "English",
          "localeId": 1033,
        },
      ]
    }
    metaData={Object {}}
    multiLanguageLocale=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    style={Object {}}
    translate={[Function]}
  />,
  "value": null,
}
`;
